app/models/widget.js

```diff
@@ -20,17 +20,45 @@ export default Model.extend({
     description: attr('string'),
     widgetType: attr('string'),
     defaultValue: attr('string'),
+    index: attr('number'),
+    parameterAliases: hasMany('parameter-alias', {
+        inverse: 'widget'
+    }),
+
+    //parameterStub: belongsTo('parameter-stub', {
+    //    inverse: 'widgets'
+    //}),
 
-    parameterMapping: hasMany('parameter-mapping', {
-        inverse: 'widget',
+    workflow: belongsTo('workflow', {
+        inverse: 'widgets'
     }),
 
-    parameters: Ember.computed('parameterMapping.@each', function() {
-        return this.get('parameterMapping')
-            .reduce((parameters, mapping) => {
-                parameters[mapping.get('mappingKey')] = mapping.get('parameter')
-                return parameters
-            }, {});
+    section: belongsTo('section', {
+        inverse: 'widgets'
     }),
 
+    _parameters: {},
+    parameters: Ember.computed('parameterAliases.@each.parameter', {
```

Comment:
Outside of the models, we never need to touch aliases, this computed takes care of it for us. It makes an object we can do lookups on to get the parameters without needing to know their real names. the same parameter can be accessed from multiple widgets this way.

Reply:
I think we might be able to do some improvements here by sideloading everything else it needs and specify the relationship as sync, but i haven't had time to play with the jsonapiserializers on the backend enough to jump down that rabbit hole yet.